src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs

```diff
@@ -129,10 +133,11 @@ protected override void ExecuteCore()
                     !string.IsNullOrEmpty(RuntimeIdentifier) &&
                     !string.IsNullOrEmpty(knownFrameworkReference.RuntimePackNamePatterns))
                 {
+                    var runtimeGraph = new RuntimeGraphCache(this).GetRuntimeGraph(RuntimeGraphPath);
                     foreach (var runtimePackNamePattern in knownFrameworkReference.RuntimePackNamePatterns.Split(';'))
                     {
                         string runtimePackRuntimeIdentifier = NuGetUtils.GetBestMatchingRid(
-                            new RuntimeGraphCache(this).GetRuntimeGraph(RuntimeGraphPath),
+                            runtimeGraph,
                             RuntimeIdentifier,
                             knownFrameworkReference.RuntimePackRuntimeIdentifiers.Split(';'),
```

Comment:
As long as you're adressing feedback from previous PR at the same time, I asked about hoisting this split too.

Reply:
But I think the whole RID matching thing needs some perf work, so feel free to leave this alone.